tools/src/main/scala/scala/scalanative/linker/Reach.scala

```diff
@@ -738,4 +772,10 @@ object Reach {
     reachability.process()
     reachability.result()
   }
+
+  private[scalanative] case class NonReachablePosition(path: Path, column: Int)
+
+  case class ReachabilityFailureException(msg: String)
+      extends RuntimeException(msg)
+      with NoStackTrace
```

Comment:
I don't see a reason for `NoStackTrace` here. `NoStackTrace` is useful for exceptions thrown in non-failure paths, which must therefore be fast to initialize. This is not the case here.

Perhaps you wanted to avoid the stack trace to be displayed in the sbt shell? If that's the case, then there is a better solution: the sbt plugin should catch the exception and wrap it in a `MessageOnlyException`.

Reply:
Also, consider extending `Exception` rather than `RuntimeException`. Or better: don't we have a generic `LinkingException` somewhere? If not, maybe we should define one.